webboot/wget_test.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+	"testing"
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name    string
+		isolink string
+		md5link string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			r, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
+			}
+
+			body, err := ioutil.ReadAll(r)
+			if err != nil {
+				t.Errorf("Failed to read the ISO: got %v, want nil", err)
+			}
+			//computes the hash of the ISO & converts it into a string to compare with md5.txt hash
+			md := md5.Sum(body)
+			hash := hex.EncodeToString(md[:])
+
+			c, err := wget(v.md5link)
+
+			var bytes []byte
+			if bytes, err = ioutil.ReadAll(c); err != nil {
+				t.Errorf("Failed to read md5.txt: got %v, want nil", err)
```

Comment:
Continue should not work. If there is an error, then bytes would most likely be nil as discussed above in another comment with "body" and therefore if bytes is nil, then the comparsion done afterwards would fail as well and not be accurate. This is why I believe a Fatalf would be better here because we should stop the user from continuing on with the rest of the code after this point.